compiler/passes/src/common/symbol_table/function_symbol.rs

```diff
@@ -34,10 +34,20 @@ pub struct FunctionSymbol {
     pub(crate) _span: Span,
     /// The inputs to the function.
     pub(crate) input: Vec<Input>,
-    /// Future inputs.
-    pub(crate) future_inputs: Vec<Location>,
-    /// The finalize block associated with the function.
-    pub(crate) finalize: Option<Location>,
+    /// The finalizer associated with this async transition.
+    pub(crate) finalize: Option<Finalizer>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
```

Comment:
Is it correct that the `Finalizer` is populated differently for an async transition vs async function?